### PR TITLE
Chore: Migrate java clients from swagger-codegen to openapi-generator

### PR DESCRIPTION
**Pull request by yue9944882**
/cc @brendandburns 

/hold
(until 5.0.0 released)

note that this pull will involve some critical dependency changes in the java repo.

### CODE REVIEW

**Comment by brendandburns** [comment]
@yue9944882 the last time I tried this, it broke a bunch of stuff because the `Quantity` and other type overrides didn't work correctly. Have you tried compiling this in the Java repo? Does it work?

Before we do this switch, I want to see the PR that it generates for the Java repo....

Thanks

**Comment by yue9944882** [comment]
@brendandburns i also tried locally but didn't other surprises except for bunch of underlying dependency switching, not sure if it's breaking backward-compatibility. i will sent a pull in the repo to reflect the generator changes.

**Comment by brendandburns** [comment]
To be clear it was the custom mappings like you added in https://github.com/kubernetes-client/gen/pull/118 that broke. So if those continue to work correctly, then this LGTM.

File: openapi/java.xml [hunk 1]
```diff
@@ -37,18 +38,16 @@
...
                 <fullJavaUtil>false</fullJavaUtil>
                 <hideGenerationTimestamp>false</hideGenerationTimestamp>
                 <dateLibrary>joda</dateLibrary>
                 <useRxJava>false</useRxJava>
                 <library>okhttp-gson</library>
-                <type-mappings>intstr.IntOrString=IntOrString,resource.Quantity=Quantity</type-mappings>
-                <import-mappings>
-                  IntOrString=io.kubernetes.client.custom.IntOrString,Quantity=io.kubernetes.client.custom.Quantity,V1Patch=io.kubernetes.client.custom.V1Patch
-                </import-mappings>
+                <typeMappings>int-or-string=IntOrString,quantity=Quantity</typeMappings>
```
Comment (yue9944882) [inline openapi/java.xml]: @wing328 for help

Comment (wing328) [reply openapi/java.xml]: How can I help? Can you share more details with me?

File: openapi/java.xml [hunk 2]
```diff
@@ -37,19 +38,17 @@
...
                 <dateLibrary>joda</dateLibrary>
                 <useRxJava>false</useRxJava>
                 <library>okhttp-gson</library>
-                <type-mappings>intstr.IntOrString=IntOrString,resource.Quantity=Quantity</type-mappings>
-                <import-mappings>
-                  IntOrString=io.kubernetes.client.custom.IntOrString,Quantity=io.kubernetes.client.custom.Quantity,V1Patch=io.kubernetes.client.custom.V1Patch
-                </import-mappings>
+                <useReflectionEqualsHashCode>true</useReflectionEqualsHashCode>
               </configOptions>
+              <typeMappings>int-or-string=IntOrString,quantity=Quantity</typeMappings>
```
Comment (wing328) [inline openapi/java.xml]: @yue9944882 I noticed that you've added https://github.com/kubernetes-client/gen/pull/117/files#diff-4b08c5154861f4aefe8a9bc110c4a15bR155 to map quantity to Quantity so I think we can remove `,quantity=Quatity` here (but I've not tested it)

Comment (fabiokung) [reply openapi/java.xml]: this is still necessary. The code you linked is patching the swagger spec to add a format field to the `Quantity` definition, so we can install a custom mapping here.

Comment (wing328) [reply openapi/java.xml]: 👌

**Comment by brendandburns** [comment]
@yue9944882 is this up to date? Now that we've merged the code changes in the java repo, I'd like to merge the corresponding generation code.

Thanks!

**Comment by yue9944882** [comment]
yes, this is in ready state

**Comment by yue9944882** [comment]
/hold cancel

@brendandburns for merging

**Pull request merged**